lib/pbench/cli/agent/commands/results/__init__.py

```diff
@@ -1,94 +1,173 @@
+
+        results_webserver = self.config.get("results", "webserver")
+        if not results_webserver:
+            logger.error(
+                "No web server host configured from which we can fetch the FQDN of the host to which we copy/move results"
+            )
+            logger.debug("'webserver' key in 'results' section not set")
+
+        server_rest_url = self.config.get("results", "server_rest_url")
+        response = requests.get(f"{server_rest_url}/host_info")
+        if response.status_code not in [200, 201]:
+            logger.error(
+                "Unable to determine results host info from %s/host_info",
+                server_rest_url,
+            )
+            sys.exit(1)
+        if response.text.startswith("MESSAGE"):
+            message = response.text.split("===")[1]
+            logger.info("*** Message from sysadmins of %s:", results_webserver)
+            logger.info("***\n*** %s", message)
+            logger.info("***\n*** No local actions taken.")
+            sys.exit(1)
+        results_path_prefix = response.text.split(":")[1]
+        if not results_path_prefix:
+            logger.error(
+                "fetch results host info did not contain a path prefix: %s",
+                response.text,
+            )
             sys.exit(1)
 
         try:
-            shutil.rmtree(result_dir)
-        except OSError:
-            logger.error("rm failed to remove the %s directory hierarchy", result_dir)
+            temp_dir = tempfile.mkdtemp(
+                dir=self.config.pbench_tmp, prefix="pbench-move-results."
+            )
+        except Exception:
+            logger.error("Failed to create temporary directory")
             sys.exit(1)
 
-        runs_copied += 1
+        user = self.context.user if self.context.user else "pbench"
+        prefix = self.context.prefix if self.context.prefix else ""
+
+        runs_copied = 0
+        failures = 0
 
-    if runs_copied + failures > 0:
-        logger.debug(
-            "successfully moved %s runs, encountered %s failures", runs_copied, failures
-        )
+        for dirent in config.pbench_run.iterdir():
+            if not self.dirent.is_dir():
+                continue
+            if dirent.name.startswith("tools-") or dirent.name == "tmp":
+                continue
+            result_dir = dirent
+            mrt = MakeResultTb(result_dir, temp_dir, user, prefix, self.config, logger)
+            make_return_val = mrt.check_result_target_dir(result_dir, temp_dir)
+            if make_return_val:
+                sys.exit(1)
+            status, result_tb_name = mrt.make_result_tb()
+
+            if status == 0:
+                if os.path.exists(result_tb_name):
+                    assert (
+                        result_tb_name
+                    ), "Logic bomb!  make_result_tb() always returns a tar ball name"
+                    crt = CopyResultTb(controller, result_tb_name, self.config, logger)
```

Comment:
I think that, as the code stands, `controller` doesn't get properly initialized when the option is specified on the command line.

If you accept my suggestion of additional options for the command option definition (see...below?), then this should be `self.controller`.

Reply:
yes

Reply:
As you surmised, `self.context.controller` -- sorry for the confusion!

Reply:
Thank you.. just wanted to be clear :)